src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java

```diff
@@ -0,0 +1,99 @@
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.request.StudentUpdateRequest;
+
+/**
+ * Action: Edits details of a student in a course.
+ */
+public class UpdateStudentAction extends Action {
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return authType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return new JsonResult(Const.StatusMessages.STUDENT_NOT_FOUND_FOR_EDIT, HttpStatus.SC_NOT_FOUND);
+        }
+
+        StudentUpdateRequest updateRequest = getAndValidateRequestBody(StudentUpdateRequest.class);
+        try {
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.updateStudentCascade(
+                    StudentAttributes.updateOptionsBuilder(courseId, studentEmail)
+                            .withName(updateRequest.getName())
+                            .withNewEmail(updateRequest.getEmail())
+                            .withTeamName(updateRequest.getTeam())
+                            .withSectionName(updateRequest.getSection())
+                            .withComment(updateRequest.getComments())
+                            .build());
+
+            StringBuilder responseMessage = new StringBuilder("Student has been updated");
+            if (!student.email.equals(updateRequest.getEmail())) {
+                logic.resetStudentGoogleId(updateRequest.getEmail(), courseId);
+                boolean isSessionSummarySendEmail =
+                        getBooleanRequestParamValue(Const.ParamsNames.SESSION_SUMMARY_EMAIL_SEND_CHECK);
+
+                if (isSessionSummarySendEmail && sendEmail(courseId, updateRequest.getEmail())) {
+                    responseMessage.append(" and email sent");
+                }
+            }
+
+            return new JsonResult(responseMessage.toString());
+        } catch (EnrollException | InvalidParametersException | EntityDoesNotExistException e) {
```

Comment:
`EntityDoesNotExistException` should be caught in a separate `catch` so that you can return a `404` instead of a `400`. 

Semantically speaking, `400` indicates that the request is malformed while `404` indicates that the resource could not be found. You can have a valid request that points to a nonexistent resource.

Reply:
Yeah your right! Updated